### PR TITLE
Follow up for "Migrate from BlockVersion.soft to BlockVersion.block_type"

### PR DESCRIPTION
**Pull request by bakulf**
Fixes: mozilla/addons/issues/15111

### Description

This PR is the follow up for #22811. It does the following:

* Remove the 'soft' field
* Make 'block_type' not nullable in the model and in the DB

### Context

See #22811.

### Testing

No new tests needed.

### Checklist

- [x] Add `#ISSUENUM` at the top of your PR to an existing open issue in the mozilla/addons repository.
- [x] Successfully verified the change locally.
- [x] The change is covered by automated tests, or otherwise indicated why doing so is unnecessary/impossible.
- [ ] Add before and after screenshots (Only for changes that impact the UI).
- [ ] Add or update relevant [docs](../docs/) reflecting the changes made.

### CODE REVIEW

**Comment by willdurand** [comment]
> Fixes: [mozilla/addons/issues/15111](https://github.com/mozilla/addons/issues/15111)

This appears to be fixed already. Either the other PR shouldn't have closed as fixed the issue, or another issue is needed :)

File: src/olympia/blocklist/migrations/0038_remove_blockversion_soft.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from django.db import migrations
+import olympia.amo.fields
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('blocklist', '0037_alter_blockversion_block_type_and_more'),
+    ]
+
+    operations = [
```
[the remark this answers is not in the document]
Comment (KevinMind) [reply src/olympia/blocklist/migrations/0038_remove_blockversion_soft.py:12]: @bakulf any news on this?

**Review by KevinMind — APPROVED** [review]
I did not verify this locally but it looks good to me code wise. The migration will run on dev and stage before prod so that might be testing enough.

Probably not a terrible idea to verify locally by
1. create a BlockVersion with block_type==None and soft==true/false
2. run the migration
3. expect it passes and the value is updated correctly.

**Comment by KevinMind** [comment]
> > Fixes: [mozilla/addons/issues/15111](https://github.com/mozilla/addons/issues/15111)
> 
> This appears to be fixed already. Either the other PR shouldn't have closed as fixed the issue, or another issue is needed :)

The former. The issue should still be open.

**Pull request merged**